### PR TITLE
feat(core): shape editor settings

### PR DESCRIPTION
**Pull request by akumatus**
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/sJGviKxfE3Ap685cl5bj/cdf4f95c-15b4-4eac-be57-26974f9ca02b.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/sJGviKxfE3Ap685cl5bj/cdf4f95c-15b4-4eac-be57-26974f9ca02b.mov">录屏2024-09-05 21.20.31.mov</video>

### CODE REVIEW

**Comment by akumatus** [comment]
* **#8122** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/8122?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts [hunk 1]
```diff
@@ -31,6 +32,7 @@ export async function getDocByName(name: DocName) {
     note: (await import('./note.json')).default,
     pen: (await import('./pen.json')).default,
     shape: (await import('./shape.json')).default,
+    flow: (await import('./flow.json')).default,
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts:35]: make these parallel using Promse.all?

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx [hunk 2]
```diff
@@ -127,14 +183,212 @@
     [t]
   );
 
+  const docs = useMemo<RadioItem[]>(
```
Comment (pengx17) [inline packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx:186]: can be moved outside of React component.

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts (hunk 1 above)
Comment (akumatus) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts:35]: It seems to be concurrent already :)

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx (hunk 2 above)
Comment (akumatus) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/shape.tsx:186]: Added missing i18n for this list.

File: packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts (hunk 1 above)
Comment (pengx17) [reply packages/frontend/core/src/components/affine/setting-modal/general-setting/editor/edgeless/docs/index.ts:35]: I guess not, sequential await is loaded one after another (n + 1 will wait for n to be finished).

**Pull request merged**